### PR TITLE
XrdOss: implement proposed solution for better rename POSIX compliancy

### PR DESCRIPTION
**Pull request by gganis**
Implements enhancement #659

### CODE REVIEW

File: src/XrdOss/XrdOssRename.cc [hunk 1]
```diff
@@ -110,12 +110,11 @@ int XrdOssSys::Rename(const char *oldname, const char *newname,
...
-// Make sure that the target file does not exist
+// If on a remote FS or symlink, make sure that the target path does not exist
 //
-#if 0
-   retc2 = lstat(local_path_New, &statbuff);
-   if (!retc2) return -EEXIST;
-#endif
+   if (!(retc2 = lstat(local_path_New, &statbuff)))
+      { if (remotefs || (statbuff.st_mode & S_IFMT) == S_IFLNK) return -EEXIST;
+      }
```
Comment (abh3) [inline src/XrdOss/XrdOssRename.cc]: What this means is that any symlinked FS will not allow a rename on top of an existing link. Is that what you really want? That excludes any virtualized filesystem even if local.

Comment (gganis) [reply src/XrdOss/XrdOssRename.cc]: The symlink part comes from your original comment. In my case I am mostly interested to local file, no symlinks. However, I see that the man page of rename says that

if newpath refers to a symbolic link, the link will be overwritten

So I suggest that we only leave the check on 'remotefs'. What do you think?

**Comment by abh3** [comment]
Well, unfortuantely, in a symlinked FS that is not a simple operation. The 
symlink refers to a data file and the data file has an xattr that records 
its symlink name. All of that has to change to successfully overwrite the 
symlink. Merely removing the symlink test would really screw things up for 
many people. I guess that's why we forbid renaming on top of an existing 
file. I don't have a problem of restricting destructive renames to simple 
file systems as long as that's OK with you.

Andy

**Comment by gganis** [comment]
>  I don't have a problem of restricting destructive renames to simple
file systems as long as that's OK with you.
Tha's ok with me. Does it mean that the PR can stay as it is?
Gerri

**Comment by abh3** [comment]
Yes, I'll merge it.

Andy

**Pull request merged**